### DISCUSSION TITLE
"Make effect parameters inline" option

### DISCUSSION THREAD

**Discussion by rampaa** [Blinue/Magpie#556] [Q&A] [answered]
"Make effect parameters inline" option states that effects must be recompiled each time their parameters are changed. Is the recompilation done by Magpie automatically?

**Comment by Blinue** [comment, marked as answer]
Yes. When applying effects for the first time, they must first be compiled into CSO, which can take several seconds for some effects. Once compiled, they are stored in the cache so they don't need to be compiled again in the future.

By default, parameters are passed to the shaders at runtime. Enabling this option, however, will inline the parameters during compilation, which can help the shader compiler optimize more effectively and reduce the number of parameters passed at runtime, leading to improved performance. The downside of this option is that cache and parameters are bound together, so changing parameters will cause the cache to become invalid and require recompilation.

These are all handled automatically, and what users may notice is that enabling this option may cause a slightly slower fullscreen entry when effect parameters are changed.
